service/src/main/java/bio/terra/catalog/datarepo/DatarepoService.java

```diff
@@ -54,6 +56,15 @@ public Map<String, List<String>> getSnapshotIdsAndRoles(AuthenticatedUserRequest
     }
   }
 
+  public SnapshotModel getPreviewTables(AuthenticatedUserRequest user, String snapshotId) {
+    try {
+      UUID id = UUID.fromString(snapshotId);
+      return snapshotsApi(user).retrieveSnapshot(id, List.of(SnapshotRetrieveIncludeModel.TABLES));
+    } catch (ApiException e) {
+      throw new DatarepoException(e.getMessage(), e.getCause());
```

Comment:
I forgot to comment on this before; in general I think it's best to include the caught exception when rethrowing. By rethrowing using `.getCause()` the stack crawl won't include the intermediate throw that resulted in `ApiException`. Is that intentional? I would expect to see instead
```suggestion
      throw new DatarepoException(e);
```

(Note that doing this will require adding another constructor to `DatarepoException`)

It also occurs to me that we'd want to make sure that this correctly maps the http result code on an access failure. I _think_ this can be done like:

```
throw new ErrorReportException(e, HttpStatus.resolve(e.getCode()));
```